### DISCUSSION TITLE
OpenSSL modules path on x64-linux-dynamic, vcpkg integrate, vcpkg env

### DISCUSSION THREAD

**Discussion by as-ascii** [microsoft/vcpkg#36715] [Q&A]
Dear All,

PoDoFo library (included in official repository) uses OpenSSL internally. It loads "legacy" OpenSSL module on runtime, using OSSL_PROVIDER_load function.

In some Linux environments everything works fine. In other environments like "clean" ubuntu docker image, OSSL_MODULES environment variable is required to be set to $VCPKG_ROOT/installed/<triplet>/lib/ossl-modules where legacy.so library is located.

Could we please discuss this issue? I would say this environment variable should be set via "vcpkg env" or "vcpkg integrate" but... it seems that commands are not working at all on Linux systems as expected by description.

Best regards.

**Comment by NieLuc** [comment]
When using Windows and trying to build a release version the libcrypto.dll links against $VCPKG_ROOT/packages/openssl_x64-windows/bin/legacy.dll. When building the legacy.dll is being copied into my project to $PROJEC_ROOT\vcpkg_installed\x64-windows\bin but not linked against that library.
